pkg/runtime/wfengine/wfengine_test.go

```diff
@@ -714,6 +715,60 @@ func TestRaiseEvent(t *testing.T) {
 	}
 }
 
+// TestPurge verifies that a workflow can have a series of activites created and then
+// verifies that all the metadata for those activities can be deleted from the statestore
+func TestPurge(t *testing.T) {
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("ActivityChainToPurge", func(ctx *task.OrchestrationContext) (any, error) {
+		val := 0
+		for i := 0; i < 10; i++ {
+			if err := ctx.CallActivity("PlusOne", val).Await(&val); err != nil {
+				return nil, err
+			}
+		}
+		return val, nil
+	})
+	r.AddActivityN("PlusOne", func(ctx task.ActivityContext) (any, error) {
+		var input int
+		if err := ctx.GetInput(&input); err != nil {
+			return nil, err
+		}
+		return input + 1, nil
+	})
+
+	ctx := context.Background()
+	client, engine, stateStore := startEngineAndGetStore(ctx, r)
+	for _, opt := range GetTestOptions() {
+		t.Run(opt(engine), func(t *testing.T) {
+			id, err := client.ScheduleNewOrchestration(ctx, "ActivityChainToPurge")
+			if assert.NoError(t, err) {
+				metadata, err := client.WaitForOrchestrationStart(ctx, id)
```

Comment:
Rather than waiting for the orchestration to start, wait for it to complete.

```suggestion
				metadata, err := client.WaitForOrchestrationCompletion(ctx, id)
```